test/parallel_api/ranges/equal_ranges_sycl.pass.cpp

```diff
@@ -0,0 +1,62 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include <oneapi/dpl/execution>
+
+#include "support/pstl_test_config.h"
+
+#if _ENABLE_RANGES_TESTING
+#include <oneapi/dpl/ranges>
+#endif
+
+#include "support/utils.h"
+
+#include <iostream>
+
+int32_t
+main()
+{
+#if _ENABLE_RANGES_TESTING
+    constexpr int max_n = 10;
+    int data1[max_n]     = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+    int data2[max_n]     = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
+    int data3[max_n]     = {0, 1, 2, -1, 4, 5, 6, 7, 8, 9};
+
+    bool res1 = false, res2 = false;
+    using namespace oneapi::dpl::experimental::ranges;
+    {
+        sycl::buffer<int> A(data1, sycl::range<1>(max_n));
+        sycl::buffer<int> B(data2, sycl::range<1>(max_n));
+        sycl::buffer<int> C(data3, sycl::range<1>(max_n));
+
+        auto view = views::all(A);
+                          
+        auto exec = TestUtils::default_dpcpp_policy;
+        using Policy = decltype(exec);
+        auto exec1 = TestUtils::make_new_policy<TestUtils::new_kernel_name<Policy, 0>>(exec);
+        auto exec2 = TestUtils::make_new_policy<TestUtils::new_kernel_name<Policy, 1>>(exec);
+             
+        res1 = equal(exec1, view, B);
+        res2 = equal(exec2, C, view, ::std::equal_to<>{});
```

Comment:
Just to think. Will it be better for coverage to test with something more complicated than `std::equal_to`, e.g. lambda that does the same: [](auto x, auto y){return x == y; }? Just to not follow the default way.

Reply:
I think that is doesn't matter... The main thing - a "call pattern" which is an equivalent of `bool operator()(const T &lhs, const T &rhs) const `.
Indeed, what cases does `[](auto x, auto y){return x == y; }` coverage and `std::equal_to` doesn't coverage?

Reply:
`equal_to` potentially can be more optimized by implementation, but lambda is something generic and can be handled differently by implementation.

Reply:
Doe you mean that probably we can missed passing comparator further to the implementation by passing std::equal instead?
In that case, yes - to cover such case we should pass another lambda/functor but with the other logic, not `{return x == y; }` :)

Reply:
Sorry for the delay. Yes, I mean that. Of course, this suggestion doesn't block the patch from merging